crates/database/src/orders.rs

```diff
@@ -560,7 +561,8 @@ array(Select (p.target, p.value, p.data) from interactions p where p.order_uid =
     where eth_o.uid = o.uid limit 1) as ethflow_data,
 (SELECT onchain_o.sender from onchain_placed_orders onchain_o where onchain_o.uid = o.uid limit 1) as onchain_user,
 (SELECT onchain_o.placement_error from onchain_placed_orders onchain_o where onchain_o.uid = o.uid limit 1) as onchain_placement_error,
-COALESCE((SELECT SUM(surplus_fee) FROM order_execution oe WHERE oe.order_uid = o.uid), 0) as executed_surplus_fee,
+COALESCE((SELECT SUM(total_fee) FROM order_execution oe WHERE oe.order_uid = o.uid), 0) as executed_total_fee,
```

Comment:
@cowprotocol/backend this will be a bug in case of multiple fulfillments of partially fillable orders where some fills are executed with fee in sell token while others are executed with fee in surplus token .
Should I have in API:
executed_total_fee[]
executed_total_fee_token[]

to fix this, or should we just live with this error for these orders that are filled during a switching period?

Reply:
The error case you are worried about is this, right?
1. partial fill for order `O` (fee stored in sell token)
2. merge PR to compute all fees in surplus token
3. another partial fill for `O` (fee stored in surplus token)
4. fetch `total_fee` for `O` results in a mix of fees from surplus and sell token

How big of a change is storing everything in surplus token? If it's not too much we could merge this and the other PR in the same week so we only have the issue on staging. I think I would be fine with that.

Alternatively would it be possible to go back over the corrupted data and patch it up?

Reply:
> The error case you are worried about is this, right?

Yes.

> How big of a change is storing everything in surplus token? If it's not too much we could merge this and the other PR in the same week so we only have the issue on staging.

We still have the same issue on prod since at the time we apply the weekly release, there are open partially fillable orders that had some fulfillments with fee in sell token and will probably have more after weekly release with fee in surplus token.

>  Alternatively would it be possible to go back over the corrupted data and patch it up?

Not impossible, but I don't see any easy way to do it, since we don't have uniform clearing prices for those trades in db (we need web3 data for it or maybe solver competition table). But such a sql migration is probably complicated and error prone. Let me try to compile it.

Reply:
Other alternative is this:
1. Merge this pr and pr for switching to fee in surplus token
2. Sql migration to go over order_executions for each order where there are entries that differ in token
3. For those, set token to surplus (or to the entry with the highest auction_id aka latest), but also set amount to be equal to protocol fee sum (which is already in surplus token). 

This way we set network fee to 0 for partial fils that were done before release, which is somewhat acceptable due to log gas fees recently and given the number of affected rows.

Reply:
I think we definitely don't want to have executedFee[] in API, so the state of the PR as is, is fine. This bug should be tackled together as part of the switch PR.